extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/logs/OpenTelemetryLogHandler.java

```diff
@@ -52,7 +53,15 @@ protected void doPublish(ExtLogRecord record) {
         }
 
         if (record.getMessage() != null) {
-            logRecordBuilder.setBody(record.getMessage());
+            // Get the message
+            final Formatter formatter = getFormatter();
+            String logMsg;
+            if (formatter != null) {
```

Comment:
Under what circumstances would the formatter be `null`?

Reply:
here, it is always null. OpenTelemetryLogRecorder doesn't set any formatter and output is not formatted

Reply:
Have you enabled OTel Logs with `quarkus.otel.logs.enabled=true`?

Reply:
Yes. Excuse me, i was not clear. I have otel log enable and working. I was looking at code because I was not able to setup the otel log pattern like with `quarkus.log.console.format`.

The formatting as expressed in https://github.com/quarkusio/quarkus/issues/44440 is done, because `logMsg = record.getFormattedMessage();` do it.

But the `getFormatter()` will always be null. At least when I run in debug, the formatter is null and I did not find any code setting it. The OpenTelemetryLogRecorder at least to not set formatter after OpenTelemetryLogHandler instanciation as other logging recorder do (`LoggingSetupRecorder.configureSocketHandler` or `LoggingSetupRecorder.configureSyslogHandler` for example).

So it seems impossible for now to format the log sended to otel collector.

Reply:
Not sure if I understand. The logged message is just another attribute of the OTel Log entry.
Anything that you could set in with formatter is probably on the log entry already... This is structured logging.
What kind of formatting would you like to do? Can you please provide an example?

Reply:
In the otel log, the logger name is not in the structure. I have `code_namespace` that's the class where the logger is invoked, and
`log_logger_namespace:org.slf4j.impl.Slf4jLogger` but I don't have the logger name, so I was trying to work around this.

Reply:
The logger name is open for debate in the OTel Semantic Conventions: https://github.com/open-telemetry/semantic-conventions/issues/1550

We can add a new attribute for it but it might change in the future. Also, most times `code_namespace` and `logger_name`will have the same values.

Will create a PR.

Reply:
created this: https://github.com/quarkusio/quarkus/pull/47454

Reply:
Cool. Thanks you. :)